vshard/util.lua

```diff
@@ -286,6 +286,14 @@ local function fiber_is_self_canceled()
     return not pcall(fiber.testcancel)
 end
 
+--
+-- Check if connection's fiber was explicitly killed
+--
+local conn_fiber_killed = function(conn)
```

Comment:
Well, probably approach is seems to be OK.
Probably you should rename function to `is_conn_fiber_killed` to conform tarantool coding conversion.